src/main/java/org/folio/circulation/domain/policy/library/EndOfCurrentHoursStrategy.java

```diff
@@ -23,9 +25,37 @@ public EndOfCurrentHoursStrategy(DateTime currentTime, DateTimeZone zone) {
     if (currentTimeInterval == null) {
       return failed(failureForAbsentTimetable());
     }
+    if(hasLibraryRolloverWorkingDay(libraryTimetable, requestedInterval)){
+      return succeeded(requestedInterval.getPrevious().getEndTime());
+    }
     if (currentTimeInterval.isOpen()) {
       return succeeded(currentTimeInterval.getEndTime());
     }
     return succeeded(currentTimeInterval.getNext().getEndTime());
   }
+
+  public static final LocalTime END_OF_A_DAY = LocalTime.MIDNIGHT.minusSeconds(1);
+
+  private boolean hasLibraryRolloverWorkingDay(LibraryTimetable libraryTimetable,
+                                               LibraryInterval requestedInterval) {
+
+    if(isNotSequenceOfWorkingDays(libraryTimetable, requestedInterval)) {
+      return false;
+    }
+
+    LocalTime endLocalTime = libraryTimetable.getHead().getEndTime().toLocalTime();
+    LocalTime startLocalTime = requestedInterval.getPrevious().getStartTime().toLocalTime();
+
+    return isDateEqualToRange(endLocalTime, LocalTime.MIDNIGHT.minusMinutes(1))
```

Comment:
Why is this one minute before midnight? Is this because the precision of the opening hours calendar is one minute?

Reply:
mod-calendar holds time in string format HH:mm, this is a reason why we use 23:59